admin/c2cgeoportal_admin/views/layertree.py

```diff
@@ -0,0 +1,73 @@
+from pyramid.httpexceptions import HTTPNotFound
+from pyramid.response import Response
+from pyramid.view import view_config
+
+from sqlalchemy.orm import subqueryload
+
+from c2cgeoportal_commons.models.main import \
+    LayergroupTreeitem, Theme, TreeGroup, TreeItem
+
+
+class LayerTreeViews():
+
+    def __init__(self, request):
+        self._request = request
+        self._dbsession = request.dbsession
+
+    @view_config(route_name="layertree",
+                 renderer="../templates/layertree.jinja2")
+    def index(self):
+        # Preload treegroups with children relations in dbsession.identity_map.
+        # Note that session identity_map keep weak references so we need to keep strong references.
+        groups = self._dbsession.query(TreeGroup). \
+            options(subqueryload("children_relation")). \
+            all()
+
+        # Preload all treeitems
```

Comment:
redundant coment ?

Reply:
why redondant, the first says that I preload treegroup with children relations and explain why, the second says that I also preload all the treeitems.

Reply:
Something like:

groups, items, themes = preloadAndGetStrongRef()
----
deleteStrongRef(groups, items, themes)

could be an alternative to comments ?

easy with python as:

>>> def test():
...     return 1,2
... 
>>> x,y = test()
>>> x
1
>>> y
2
>>>